public/app/features/alerting/unified/components/rule-editor/labels/LabelsField.tsx

```diff
@@ -0,0 +1,459 @@
+  labelsInSubform: Array<{ key: string; value: string }>;
+};
+
+export interface LabelsSubFormProps {
+  dataSourceName: string;
+  onClose: () => void;
+}
+
+export function LabelsSubForm({ dataSourceName, onClose }: LabelsSubFormProps) {
+  const styles = useStyles2(getStyles);
+  // get initial values from the parent form
+  const { setValue, getValues } = useFormContext<RuleFormValues>();
+  const initialLabels = getValues('labels');
+
+  const onSave = (labels: LabelsSubformValues) => {
+    // Save the labels to the parent form
+    setValue('labels', labels.labelsInSubform);
+    onClose();
+  };
+  // default values for the subform are the initial labels
+  const defaultValues: LabelsSubformValues = useMemo(() => {
+    return { labelsInSubform: initialLabels };
+  }, [initialLabels]);
+  const formAPI = useForm<LabelsSubformValues>({ defaultValues });
+  return (
+    <FormProvider {...formAPI}>
+      <form onSubmit={formAPI.handleSubmit(onSave)}>
+        <Stack direction="column" gap={1} alignItems="center" justifyContent="center">
+          <LabelsWithSuggestions dataSourceName={dataSourceName} />
+          <Space v={1} />
+          <LabelsInRule labels={formAPI.watch('labelsInSubform')} />
+          <Space v={1} />
+          <div className={styles.confirmButton}>
+            <Button type="button" variant="secondary" onClick={onClose}>
+              Cancel
+            </Button>
+            <Button type="submit">Confirm</Button>
+          </div>
+        </Stack>
+      </form>
+    </FormProvider>
+  );
+}
+export function useCombinedLabels(
+  dataSourceName: string,
+  labelsPluginInstalled: boolean,
+  loadingLabelsPlugin: boolean
+) {
+  // ------- Get labels keys and their values from existing alerts
+  const { loading, labelsByKey: labelsByKeyFromExisingAlerts } = useGetCustomLabels(dataSourceName);
+  // ------- Get only the keys from the ops labels, as we will fetch the values for the keys once the key is selected.
+  const { loading: isLoadingLabels, labelsOpsKeys = [] } = useGetOpsLabelsKeys(
+    !labelsPluginInstalled || loadingLabelsPlugin
+  );
+  //------ Convert the labelsOpsKeys to the same format as the labelsByKeyFromExisingAlerts
+  const labelsByKeyOps = useMemo(() => {
+    return labelsOpsKeys.reduce((acc: Record<string, Set<string>>, label) => {
+      acc[label.name] = new Set();
+      return acc;
+    }, {});
+  }, [labelsOpsKeys]);
+
+  //------- Convert the keys from the ops labels to options for the dropdown
+  const keysFromGopsLabels = useMemo(() => {
+    return mapLabelsToOptions(Object.keys(labelsByKeyOps));
+  }, [labelsByKeyOps]);
+
+  //------- Convert the keys from the existing alerts to options for the dropdown
+  const keysFromExistingAlerts = useMemo(() => {
+    return mapLabelsToOptions(Object.keys(labelsByKeyFromExisingAlerts));
+  }, [labelsByKeyFromExisingAlerts]);
+
+  // create two groups of labels, one for ops and one for custom
+  const groupedOptions = [
```

Comment:
nit: for grouped options we should probably set `expanded: true` so the categories are opened by default, feels a bit weird to have to choose which one to open when I'm just looking for a value :)